### PR TITLE
Added reinit__is_reduced and sync_all_reduce docs in metrics doc

### LINKED ISSUE #1315: Fix link to reinit__is_reduced, sync_all_reduce
## 🚀 Feature

Quick issue to add the links to `reinit__is_reduced` and `sync_all_reduce`  in the docs:  https://pytorch.org/ignite/metrics.html#metrics-and-distributed-computations

In addition, please check if `reinit__is_reduced` and `sync_all_reduce` are used elsewhere in the docs and do not contain a link to their docs.

For Hacktoberfest contributors, feel free to ask questions for details if any and say that you would like to tackle the issue.
Please, take a look at CONTRIBUTING guide.

### PR DESCRIPTION
**Pull request by josselineperdomo**
Fixes #1315  

Description:
Added links to `reinit__is_reduced` and `sync_all_reduce` decorators into `ignite.metrics` documentation.

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [x] Documentation is updated (if required)

### CODE REVIEW

File: docs/source/metrics.rst [hunk 1]
```diff
@@ -256,12 +256,15 @@ Complete list of metrics
     - :class:`~ignite.metrics.MetricsLambda`
     - :class:`~ignite.metrics.Precision`
     - :class:`~ignite.metrics.Recall`
+    - :meth:`~ignite.metrics.metric.reinit__is_reduced`
```
Comment (vfdev-5) [inline docs/source/metrics.rst:259]: actually, these methods are already added to the docs in the end, see here lines 326 and 328.

**Review by vfdev-5 — COMMENTED** [review]
Looks good, thanks @josselineperdomo  !
I left a comment to improve the PR

File: docs/source/metrics.rst (hunk 1 above)
Comment (josselineperdomo) [reply docs/source/metrics.rst]: @vfdev-5 I already updated the order in the metric list :)

Comment (vfdev-5) [reply docs/source/metrics.rst]: @josselineperdomo ops, sorry, I misunderstood what was done. However, `reinit__is_reduced` and `sync_all_reduce` are not metrics but just helpers. So, let's remove them from the list and keep the order as it was:
```
.. autofunction:: sync_all_reduce

.. autofunction:: reinit__is_reduced
```

Comment (josselineperdomo) [reply docs/source/metrics.rst]: Sorry! I misunderstood and I thought I had to put in the metric list the decorators. My last commit fixed it

**Review by vfdev-5 — APPROVED** [review]
LGTM! Thanks @josselineperdomo

**Comment by josselineperdomo** [comment]
> LGTM! Thanks @josselineperdomo

No worries :blush: I open to work on another issue, I'm a beginner contributor and I have some experience working with Pytorch

**Comment by vfdev-5** [comment]
@josselineperdomo sounds good, please check some unassigned issues from here : https://github.com/pytorch/ignite/issues?q=is%3Aissue+is%3Aopen+label%3AHacktoberfest+no%3Aassignee

There certainly can be one that you'd like to work on :)

**Pull request merged**